### PR TITLE
feat: add searchbar for mobile devices (new design)

### LINKED ISSUE #2202: feat: add searchbar for mobile devices (new design)
> You can view the figma file [here](https://www.figma.com/file/ntisIxc9QHwOZ828CH5G3a/LinksHub?type=design&node-id=17-231&mode=design&t=29U3Qw0LktnVRteR-0).

image
image

> The design should be responsive. 

---

ℹ️ We recommend updating the styling of existing components, otherwise you can add extra components if there is a need.

- The nav menu is implemented in `dev` branch, so make sure to fork the `dev` branch as well. Then you can use the existing code easily.

> Any changes to the above design are welcome. Please feel free to share your thoughts, and we will discuss them thoroughly.

---

### PR DESCRIPTION
**Pull request by heysolomon**
## Fixes Issue
Closes #2202

## Changes proposed
- added the new design for the searchbar

## Screenshots
image

### CODE REVIEW

**Comment by Anmol-Baranwal** [comment]
@rupali-codes 
Vercel deployment :)

**Review by rupali-codes — COMMENTED** [review]
@Anmol-Baranwal the deployment is failing 

image

**Comment by heysolomon** [comment]
> @Anmol-Baranwal the deployment is failing
> 
> image

Sorry is it from my end?

I'm I supposed to fix something?

**Comment by rupali-codes** [comment]
> > @Anmol-Baranwal the deployment is failing
> > image
> 
> Sorry is it from my end?
> 
> I'm I supposed to fix something?

lemme try helping you

**Comment by heysolomon** [comment]
> > > @Anmol-Baranwal the deployment is failing
> > > image
> > 
> > 
> > Sorry is it from my end?
> > I'm I supposed to fix something?
> 
> lemme try helping you

Okay, thank you!

File: components/SideNavbar/SideNavbarCategoryList.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ export const SideNavbarCategoryList: FC<{
     <ul
       data-custom="restrict-click-outside"
       ref={listRef}
-      className="relative w-full max-h-full flex flex-col sc overflow-y-scroll purpleScrollbar pr-4"
+      className="relative w-full max-h-full flex flex-col sc overflow-y-scroll scrollColorDarkMobile pr-4"
```
Comment (aftabrehan) [inline components/SideNavbar/SideNavbarCategoryList.tsx:35]: The white scrollbar isn't visible in light mode.

Screenshot 2024-01-03 at 10 52 27 AM


```suggestion
      className="relative w-full max-h-full flex flex-col sc overflow-y-scroll purpleScrollbar dark:scrollColorDarkMobile pr-4"
```

I'm not sure if this suggestion would work, but please make sure to test both light and dark themes.

**Review by aftabrehan — CHANGES_REQUESTED** [review]
Excellent PR, @heysolomon 💪 🚀 💯 

It just needs minor adjustments and then we're good to go 🚀

**Comment by rupali-codes** [comment]
@heysolomon still working on it?

**Comment by heysolomon** [comment]
> @heysolomon still working on it?

yes

File: components/SideNavbar/SideNavbarCategoryList.tsx (hunk 1 above)
Comment (heysolomon) [reply components/SideNavbar/SideNavbarCategoryList.tsx:35]: Okay. Let me fix it.

Comment (heysolomon) [reply components/SideNavbar/SideNavbarCategoryList.tsx:35]: I was able to fix the issue.

**Review by aftabrehan — APPROVED** [review]
Looks good to me! 👍

**Comment by aftabrehan** [comment]
The PR is okay, @rupali-codes, @Anmol-Baranwal, and @CBID2 please test on your end and help merge the PR.

**Review by Anmol-Baranwal — APPROVED** [review]
🚀 Looks good to me. 

Works as mentioned :)

> @rupali-codes @CBID2 
Review it.

**Review by rupali-codes — APPROVED** [review]
looks perfect to me

Thanks @heysolomon

**Pull request merged**